src/internal/storage/chunk/storage.go

```diff
@@ -81,9 +82,12 @@ func (s *Storage) PrefetchData(ctx context.Context, dataRef *DataRef) error {
 
 // List lists all of the chunks in object storage.
 func (s *Storage) List(ctx context.Context, cb func(id ID) error) error {
-	return errors.EnsureStack(s.store.Walk(ctx, nil, func(key []byte) error {
-		return cb(ID(key))
-	}))
+	it := s.store.NewKeyIterator(kv.Span{})
+	return stream.ForEach(ctx, it, func(key []byte) error {
+		var id ID
+		copy(id[:], key)
```

Comment:
Is the key the chunk itself? I am assuming that the key is an index and the value is the chunk when reading this. Do we want to be executing the callback against the key or the value?

Reply:
Yeah you're right.  It's not the chunk id, it's the whole key (which includes the chunk prefix).  I need to fix this.  We have to parse the key and discard the chunk part.